crates/superposition_proc_macros/src/lib.rs

```diff
@@ -0,0 +1,48 @@
+extern crate proc_macro;
```

Comment:
We already have a macros crate right? Can we group all macros in one place?

Reply:
@ayushjain17 should we use proc_macros here? Why not directly implement `ToSql` and `FromSql` traits on a struct?

If the argument is that we need this later, can we add it when we need it? I don't want to prematurely optimize

Reply:
proc_macros can be in a crate which are used to write proc_macros only, you cannot export other macros from that crate, it stops compiling with that error itself

Reply:
> can we add it when we need it? I don't want to prematurely optimize

it is already added now, so it would basically be like, remove the optimisation which is already added and then add it later on again when we need it

and mostly, we forget to catch and reduce these redundancies later on when they occur and then we add more work to just add refactor to optimise these

when we can already solve it when we have already identified it as a future use-case

also, it mainly doesn't make sense to un-optimise something which has already been optimised

Reply:
> it is already added now, so it would basically be like, remove the optimisation which is already added and then add it later on again when we need it

Its an unoptimization right now, given the added compile time for macros. It's not used anywhere else. Why do we need it as a proc_macro instead of a direct trait implementation?

Reply:
@Datron , I think the intention here is to put up an easy way for other people who will need to do something similar in future. @ayushjain17 has already figured out the solution to this kind of problem and put it in place, this will helpful.

And for the compile time argument, yes this will increase, but our project already uses a lot of derive macros, the delta here will be minimal, we won't be anyway writing that many macros.

Reply:
![image](https://github.com/user-attachments/assets/8247fee3-5917-43c9-96fb-f8bd4b177ad2)
These are the results of profiling . The above is for main which takes 81.7 seconds 
This one is from ayush's branch , it takes 80.4 seconds : 
![image](https://github.com/user-attachments/assets/aeb0e496-9c38-4c27-9104-c7cde7e30cf7)
Now , we can decide what to go forward with.